io/io/test/TFileTests.cxx

```diff
@@ -1,7 +1,67 @@
-#include "TFile.h"
+#include <vector>
 
 #include "gtest/gtest.h"
 
+#include "TFile.h"
+#include "TKey.h"
+#include "TNamed.h"
+#include "TSystem.h"
+
+TEST(TFile, WriteObjectTObject)
+{
+    auto filename{"tfile_writeobject_tobject.root"};
+    auto tnamed_name{"mytnamed_name"};
+    auto tnamed_title{"mytnamed_title"};
+
+    {
+        TNamed mytnamed{tnamed_name, tnamed_title};
+        TFile f{filename, "recreate"};
+        f.WriteObject(&mytnamed, mytnamed.GetName());
+        f.Close();
+    }
+
+    TFile input{filename};
+    auto named = input.Get<TNamed>(tnamed_name);
+
+    EXPECT_STREQ(named->GetName(), tnamed_name);
+    EXPECT_STREQ(named->GetTitle(), tnamed_title);
```

Comment:
You should also test the value of the TKey's title (the object itself has its title stored within the buffer inside the key and so was (should be) right even before this PR).

Reply:
Yes, good point. Just for reference, here's the output from ROOT 6.24/02. Indeed the object itself was already storing the correct title even before this commit:

### tfile_writeobject_printkey.cpp
```cpp

#include <iostream>

#include "TFile.h"
#include "TKey.h"
#include "TNamed.h"

int main(){

    auto filename{"tfile_writeobject_tobject.root"};
    auto tnamed_name{"mytnamed_name"};
    auto tnamed_title{"mytnamed_title"};

    {
        TNamed mytnamed{tnamed_name, tnamed_title};
        TFile f{filename, "recreate"};
        f.WriteObject(&mytnamed, mytnamed.GetName());
        f.Close();
    }

    TFile input{filename};
    auto named = input.Get<TNamed>(tnamed_name);
    auto keyptr = static_cast<TKey *>(input.GetListOfKeys()->At(0));

    std::cout << "Stored object's name: " << named->GetName() << "\n";
    std::cout << "Stored object's title: " << named->GetTitle() << "\n";
    std::cout << "Key's name: " << keyptr->GetName() << "\n";
    std::cout << "Key's title: " << keyptr->GetTitle() << "\n";
}
```

### output
```bash
$: root --version
ROOT Version: 6.24/02
Built for linuxx8664gcc on Jul 03 2021, 08:02:00
$: g++ -o tfile_writeobject_printkey tfile_writeobject_printkey.cpp `root-config --cflags --glibs`
$: ./tfile_writeobject_printkey 
Stored object's name: mytnamed_name
Stored object's title: mytnamed_title
Key's name: mytnamed_name
Key's title: object title
```